### DISCUSSION TITLE
"Insert ignore comments" feature for old codebases

### DISCUSSION THREAD

**Discussion by notramo** [biomejs/biome#1510] [Ideas]
When working on old and bad codebase, it's currently not possible to introduce strict Biome linting.
Mostly, the requirements are:

- Do not refactor old code.
- Have the strictest linting for new code.

However, with hundreds of linter errors throughout the codebase, there are two choices: refactor, and fix them (and in very bad codebases probably introduce bugs during the process) or add ignore comments. Linter error fixing is mostly pointless in bad code as it needs more refactoring anyways. Also, in most of the cases, there is no budget for it.

The second (add ignore comments) could be automated. The workflow would be:
- Fix fixable errors.
- Ask the user to apply unsafe fixes.
- Add ignore comment to silence the remaining errors (comments must be specific to the type of error encountered).
- Then, after the errors are silenced, configure Biome in [`lefthook`](https://github.com/evilmartians/lefthook) and have every new commit strictly linted.

**Comment by canadaduane** [comment]
I came here looking for this. Thanks for the nice description of the problem and solution.

Related work available for eslint:
- https://github.com/mizdra/eslint-interactive
- https://github.com/amanda-mitchell/suppress-eslint-errors

The above may serve as either a research list, or reference point for building future support for Biome.
